web/src/user_group_edit.ts

```diff
@@ -516,7 +518,109 @@ function update_membership_status_text(group: UserGroup): void {
+    const button_state = show_change_process_button ? "unsaved" : "discarded";
+    settings_components.change_save_button_state($save_button_controls, button_state);
+}
+
+function get_request_data_for_removing_group_permission(
+    current_value: GroupSettingValue,
+    removed_group_id: number,
+): string {
+    const nobody_group = user_groups.get_user_group_from_name("role:nobody")!;
+
+    if (typeof current_value === "number") {
+        return JSON.stringify({
+            new: nobody_group.id,
+            old: current_value,
+        });
+    }
+
+    let new_setting_value: GroupSettingValue = {...anonymous_group_schema.parse(current_value)};
+    new_setting_value.direct_subgroups = new_setting_value.direct_subgroups.filter(
+        (group_id) => group_id !== removed_group_id,
+    );
+
+    if (
+        new_setting_value.direct_subgroups.length === 0 &&
+        new_setting_value.direct_members.length === 0
+    ) {
+        new_setting_value = nobody_group.id;
+    }
+    return JSON.stringify({
+        new: new_setting_value,
+        old: current_value,
+    });
+}
+
+function populate_data_for_removing_realm_permissions(
+    $subsection: JQuery,
+    group: UserGroup,
+): Record<string, string> {
+    const changed_setting_elems = settings_components
+        .get_subsection_property_elements($subsection)
+        .filter((elem) => !$(elem).prop("checked"));
+    const changed_setting_names = changed_setting_elems.map((elem) => $(elem).attr("name")!);
+
+    const data: Record<string, string> = {};
+    for (const setting_name of changed_setting_names) {
+        const current_value = realm[realm_schema.keyof().parse("realm_" + setting_name)];
+        data[setting_name] = get_request_data_for_removing_group_permission(
+            group_setting_value_schema.parse(current_value),
+            group.id,
+        );
+    }
+
+    return data;
+}
+
+export function update_setting_in_group_permissions_panel(
+    $setting_elem: JQuery,
+    new_value: GroupSettingValue,
+    old_value: GroupSettingValue,
+): void {
+    if (!overlays.groups_open()) {
+        return;
+    }
+
+    const active_group_id = get_active_data().id;
+    if (active_group_id === undefined) {
+        return;
+    }
+
+    const group_had_permission_previously = user_groups.group_has_permission(
+        old_value,
+        active_group_id,
+    );
+    const group_has_permission = user_groups.group_has_permission(
+        group_setting_value_schema.parse(new_value),
+        active_group_id,
+    );
+
+    if (group_had_permission_previously === group_has_permission) {
+        return;
+    }
+
+    if (!group_has_permission) {
+        const $subsection = $setting_elem.closest(".settings-subsection-parent");
+        $setting_elem.closest(".input-group").remove();
+
+        if ($subsection.find(".input-group").length === 0) {
+            $subsection.addClass("hide");
+        }
+    }
```

Comment:
Hmm, is this missing logic to add it if the group now does have the permission?

Reply:
Yes. I have mentioned in the comments. Will work on this now.